### ISSUE TITLE
Mac OS crash on launch since 2.19.0

### ISSUE THREAD

**Issue by DarkFenX** [pyfa-org/Pyfa#2156]
Originally reported in #2146

2.19.0 on my mac:
```
Traceback (most recent call last):
  File "PyInstaller/loader/rthooks/pyi_rth_pkgres.py", line 13, in <module>
  File "/usr/local/lib/python3.7/site-packages/PyInstaller/loader/pyimod03_importers.py", line 623, in exec_module
  File "pkg_resources/__init__.py", line 86, in <module>
ModuleNotFoundError: No module named 'pkg_resources.py2_warn'
[2759] Failed to execute script pyi_rth_pkgres
```
But 2.18.1 runs fine.

**Comment by DarkFenX** [comment]
Library differences between those two builds:
```
Downloading matplotlib-3.2.0-cp37-cp37m-macosx_10_9_x86_64.whl (12.4 MB)
Downloading matplotlib-3.2.1-cp37-cp37m-macosx_10_9_x86_64.whl (12.4 MB)

Downloading SQLAlchemy-1.3.14.tar.gz (6.1 MB)
Downloading SQLAlchemy-1.3.15.tar.gz (6.1 MB)

cryptography-2.8-cp34-abi3-macosx_10_6_intel.whl (1.6 MB)
cryptography-2.9-cp35-abi3-macosx_10_9_intel.whl (1.7 MB)

Downloading PyYAML-5.3.tar.gz (268 kB)
Downloading PyYAML-5.3.1.tar.gz (269 kB)

Requirement already satisfied: six in /usr/local/Cellar/protobuf/3.11.2/libexec/lib/python3.7/site-packages (from wxPython==4.0.6->-r requirements.txt (line 1)) (1.13.0)
Requirement already satisfied: six in /usr/local/Cellar/protobuf/3.11.4/libexec/lib/python3.7/site-packages (from wxPython==4.0.6->-r requirements.txt (line 1)) (1.14.0)

Downloading Pillow-7.0.0-cp37-cp37m-macosx_10_6_intel.whl (3.9 MB)
Downloading Pillow-7.1.1-cp37-cp37m-macosx_10_10_x86_64.whl (2.2 MB)

Downloading pyparsing-2.4.6-py2.py3-none-any.whl (67 kB)
Downloading pyparsing-2.4.7-py2.py3-none-any.whl (67 kB)

Downloading kiwisolver-1.1.0-cp37-cp37m-macosx_10_6_intel.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl (113 kB)
Downloading kiwisolver-1.2.0-cp37-cp37m-macosx_10_9_x86_64.whl (60 kB)

Downloading certifi-2019.11.28-py2.py3-none-any.whl (156 kB)
Downloading certifi-2020.4.5.1-py2.py3-none-any.whl (157 kB)

Requirement already satisfied: setuptools in /usr/local/lib/python3.7/site-packages (from kiwisolver>=1.0.1->matplotlib>=3.1.2->-r requirements.txt (line 3)) (42.0.2)
Requirement already satisfied: setuptools in /usr/local/lib/python3.7/site-packages (from PyInstaller==3.6) (46.0.0)
```

**Comment by DarkFenX** [comment]
setuptools upgrade from 42.0.2 to 46.0.0 seems to cause it

**Comment by DarkFenX** [comment]
Root cause: https://github.com/pyinstaller/pyinstaller/issues/4672 + newer setuptools version bundled with build environment

Workaround implemented in 1f83dba1acd8a5443be236a92e096d7b6e79dcf5.
